File: get-wfactions-path.js
const { existsSync } = require("fs")
const { join }= require("path")
const { execFileSync } = require("child_process")

// Determine where WFActions.plist is
// Supported platforms: macOS, iOS, iSH on iOS
const wfActions = "WFActions.plist"
const wfKitPath = "/System/Library/PrivateFrameworks/WorkflowKit.framework"

switch (process.platform) {
    case "darwin":
        // Apple platform
        const iosWFActionsPath = join(wfKitPath, wfActions)
        // If WFActions.plist exists at iOS path, then environment is iOS
        // Otherwise, environment is macOS
        module.exports = existsSync(iosWFActionsPath) ? iosWFActionsPath : join(wfKitPath, "Resources", wfActions)
        break
    case "linux":
        // Assume environment is iSH on iOS for now
        const mountDirectory = "/mnt"
        execFileSync("mount", ["-t", "real", "/", mountDirectory])
        const wfActionsPath = join(mountDirectory, wfKitPath, wfActions)
        if (existsSync(wfActionsPath)) {
            // Yes, this is iSH on iOS
            module.exports = wfActionsPath
        } else {
            // Linux but not iSH, so do not proceed
            console.log("The only supported Linux environment for this script is iSH on iOS.")
            process.exit()
        }
        break
    default:
        // Unsupported OS
        console.log("This OS is not supported.")
        process.exit()
}

File: get_wfactions_path.py
import os
import platform
import subprocess

# Determine where WFActions.plist is
# Supported platforms: macOS, iOS, iSH on iOS
WFACTIONS = "WFActions.plist"
WFKIT_PATH = "/System/Library/PrivateFrameworks/WorkflowKit.framework"
OS_NAME = platform.system()

def get_path():
    '''Return the path to WFActions.plist or exit the program depending on the platform.'''
    
    if OS_NAME == "Darwin":
        ios_wfactions_path = os.path.join(WFKIT_PATH, WFACTIONS)
        if os.path.exists(ios_wfactions_path):
            # Environment is iOS
            return ios_wfactions_path
        else:
            # Environment is macOS
            return os.path.join(WFKIT_PATH, "Resources", WFACTIONS)
    elif OS_NAME == "Linux":
        # Assume environment is iSH on iOS for now
        mount_directory = "/mnt"
        subprocess.run(["mount", "-t", "real", "/", mount_directory])
        wfactions_path = os.path.join(mount_directory, WFKIT_PATH, WFACTIONS)
        if os.path.exists(wfactions_path):
            return wfactions_path
        else:
            # Linux but not iSH, so do not proceed
            print("The only supported Linux environment for this script is iSH on iOS.")
            quit()
    else:
        # Unsupported OS
        print("This OS is not supported.")
        quit()